### PR TITLE
Add blueprint language

### PR DESCRIPTION
**Pull request by seshotake**
- Add blueprint tree-sitter queries

Screenshot from 2023-06-03 10-01-42

### CODE REVIEW

File: languages.toml [hunk 1]
```diff
@@ -2577,3 +2577,16 @@ indent = { tab-width = 4, unit = "\t" }
...
+scope = "source.blueprint"
+injection-regex = "blueprint"
+file-types = ["blp"]
+roots = []
+comment-token = "//"
+indent = { tab-width = 4, unit = "    " }
+
+[[grammar]]
+name = "blueprint"
+source = { git = "https://gitlab.com/gabmus/tree-sitter-blueprint.git", rev = "7f1a5df44861291d6951b6b2146a9fef4c226e14" }
```
Comment (the-mikedavis) [inline languages.toml:2592]:
```suggestion
source = { git = "https://gitlab.com/gabmus/tree-sitter-blueprint", rev = "7f1a5df44861291d6951b6b2146a9fef4c226e14" }
```

**Review by the-mikedavis — COMMENTED** [review]
Just a minor nit, otherwise this looks good.

It looks like blueprint includes an official language server that could be added too if it works out of the box: https://jwestman.pages.gitlab.gnome.org/blueprint-compiler/#features

(Connects https://github.com/helix-editor/helix/issues/7121)

File: languages.toml (hunk 1 above)
Comment (seshotake) [reply languages.toml]: Fixed

**Comment by seshotake** [comment]
I'd like to add that the LSP  does not seem to be working correctly. I don't know if the problem is in my configuration (I just copy-pasted from https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#blueprint_ls) or with blueprint-compiler.

helix.log

```
2023-06-05T03:27:11.812 helix_lsp::transport [INFO] blueprint-compiler -> {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"general":{"positionEncodings":["utf-8","utf-32","utf-16"]},"textDocument":{"codeAction":{"codeActionLiteralSupport":{"codeActionKind":{"valueSet":["","quickfix","refactor","refactor.extract","refactor.inline","refactor.rewrite","source","source.organizeImports"]}}},"completion":{"completionItem":{"deprecatedSupport":true,"insertReplaceSupport":true,"resolveSupport":{"properties":["documentation","detail","additionalTextEdits"]},"snippetSupport":true,"tagSupport":{"valueSet":[1]}},"completionItemKind":{}},"hover":{"contentFormat":["markdown"]},"inlayHint":{"dynamicRegistration":false},"publishDiagnostics":{"versionSupport":true},"rename":{"dynamicRegistration":false,"honorsChangeAnnotations":false,"prepareSupport":true},"signatureHelp":{"signatureInformation":{"activeParameterSupport":true,"documentationFormat":["markdown"],"parameterInformation":{"labelOffsetSupport":true}}}},"window":{"workDoneProgress":true},"workspace":{"applyEdit":true,"configuration":true,"didChangeConfiguration":{"dynamicRegistration":false},"executeCommand":{"dynamicRegistration":false},"inlayHint":{"refreshSupport":false},"symbol":{"dynamicRegistration":false},"workspaceEdit":{"documentChanges":true,"failureHandling":"abort","normalizesLineEndings":false,"resourceOperations":["create","rename","delete"]},"workspaceFolders":true}},"clientInfo":{"name":"helix","version":"23.05 (1cdb64bf)"},"processId":34747,"rootPath":"/home/seshotake/Documents/Projects/FlashCards","rootUri":"file:///home/seshotake/Documents/Projects/FlashCards","workspaceFolders":[{"name":"FlashCards","uri":"file:///home/seshotake/Documents/Projects/FlashCards"}]},"id":0}
2023-06-05T03:27:11.908 helix_lsp::transport [ERROR] blueprint-compiler err <- "input: {\"jsonrpc\":\"2.0\",\"method\":\"initialize\",\"params\":{\"capabilities\":{\"general\":{\"positionEncodings\":[\"utf-8\",\"utf-32\",\"utf-16\"]},\"textDocument\":{\"codeAction\":{\"codeActionLiteralSupport\":{\"codeActionKind\":{\"valueSet\":[\"\",\"quickfix\",\"refactor\",\"refactor.extract\",\"refactor.inline\",\"refactor.rewrite\",\"source\",\"source.organizeImports\"]}}},\"completion\":{\"completionItem\":{\"deprecatedSupport\":true,\"insertReplaceSupport\":true,\"resolveSupport\":{\"properties\":[\"documentation\",\"detail\",\"additionalTextEdits\"]},\"snippetSupport\":true,\"tagSupport\":{\"valueSet\":[1]}},\"completionItemKind\":{}},\"hover\":{\"contentFormat\":[\"markdown\"]},\"inlayHint\":{\"dynamicRegistration\":false},\"publishDiagnostics\":{\"versionSupport\":true},\"rename\":{\"dynamicRegistration\":false,\"honorsChangeAnnotations\":false,\"prepareSupport\":true},\"signatureHelp\":{\"signatureInformation\":{\"activeParameterSupport\":true,\"documentationFormat\":[\"markdown\"],\"parameterInformation\":{\"labelOffsetSupport\":true}}}},\"window\":{\"workDoneProgress\":true},\"workspace\":{\"applyEdit\":true,\"configuration\":true,\"didChangeConfiguration\":{\"dynamicRegistration\":false},\"executeCommand\":{\"dynamicRegistration\":false},\"inlayHint\":{\"refreshSupport\":false},\"symbol\":{\"dynamicRegistration\":false},\"workspaceEdit\":{\"documentChanges\":true,\"failureHandling\":\"abort\",\"normalizesLineEndings\":false,\"resourceOperations\":[\"create\",\"rename\",\"delete\"]},\"workspaceFolders\":true}},\"clientInfo\":{\"name\":\"helix\",\"version\":\"23.05 (1cdb64bf)\"},\"processId\":34747,\"rootPath\":\"/home/seshotake/Documents/Projects/FlashCards\",\"rootUri\":\"file:///home/seshotake/Documents/Projects/FlashCards\",\"workspaceFolders\":[{\"name\":\"FlashCards\",\"uri\":\"file:///home/seshotake/Documents/Projects/FlashCards\"}]},\"id\":0}\n"
2023-06-05T03:27:11.909 helix_lsp::transport [INFO] blueprint-compiler <- {"id":0,"result":{"capabilities":{"textDocumentSync":{"openClose":true,"change":2},"semanticTokensProvider":{"legend":{"tokenTypes":["enumMember"]},"full":true},"completionProvider":{},"codeActionProvider":{},"hoverProvider":true},"serverInfo":{"name":"Blueprint","version":"0.8.1"}},"jsonrpc":"2.0"}

2023-06-05T03:27:11.909 helix_lsp::transport [ERROR] blueprint-compiler err <- "output: {\"id\":0,\"result\":{\"capabilities\":{\"textDocumentSync\":{\"openClose\":true,\"change\":2},\"semanticTokensProvider\":{\"legend\":{\"tokenTypes\":[\"enumMember\"]},\"full\":true},\"completionProvider\":{},\"codeActionProvider\":{},\"hoverProvider\":true},\"serverInfo\":{\"name\":\"Blueprint\",\"version\":\"0.8.1\"}},\"jsonrpc\":\"2.0\"}\r\n"
2023-06-05T03:27:11.909 helix_lsp::transport [ERROR] blueprint-compiler err <- "\n"
2023-06-05T03:27:11.909 helix_lsp::transport [INFO] blueprint-compiler <- {"capabilities":{"codeActionProvider":{},"completionProvider":{},"hoverProvider":true,"semanticTokensProvider":{"full":true,"legend":{"tokenTypes":["enumMember"]}},"textDocumentSync":{"change":2,"openClose":true}},"serverInfo":{"name":"Blueprint","version":"0.8.1"}}
2023-06-05T03:27:11.909 helix_lsp [ERROR] failed to initialize language server: failed to parse: data did not match any variant of untagged enum SemanticTokensServerCapabilities
2023-06-05T03:27:12.189 helix_view::document [DEBUG] id 2 modified - last saved: 0, current: 0
```

**Comment by the-mikedavis** [comment]
Ah ok, we can leave it out of this PR for now. That looks like an upstream issue where the language server is sending something for the semantic tokens provider capabilities that doesn't match the LSP spec.

**Comment by seshotake** [comment]
Should I revert the commit or just leave it as it is?

**Comment by the-mikedavis** [comment]
Actually it's probably ok to leave it in this PR. Once that gets resolved upstream, the language server should work fine without any more changes to Helix. There isn't really any downside of a language server breaks spec like this: it just won't be initialized so the LSP features won't work.

**Review by the-mikedavis — APPROVED** [review]

**Comment by gabydd** [comment]
don't have gnome gitlab account so posting here the reason why ``SemanticTokensServerCapabilities`` is not being deserialized is because ``legend`` requires a ``tokenModifiers`` array: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#semanticTokensLegend which is not being provided, probably should be able to just make it an empty array

**Comment by seshotake** [comment]
Thanks, @gabydd.The problem was indeed that `legend` had missing `tokenModifiers`. So, I already reported in gitlab.gnome.org and send PR to fix it to upstream

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**